.pre-commit-config.yaml

```diff
@@ -0,0 +1,28 @@
+exclude: ".*\\.asdf$"
+
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v5.0.0
+    hooks:
+      - id: check-added-large-files
+      - id: check-ast
+      - id: check-case-conflict
+      - id: check-yaml
+        args: ["--unsafe"]
+      - id: check-toml
+      - id: check-merge-conflict
+      - id: check-symlinks
+      - id: debug-statements
+      - id: detect-private-key
+      # - id: end-of-file-fixer
+      # - id: trailing-whitespace
+  - repo: https://github.com/astral-sh/ruff-pre-commit
+    rev: 'v0.9.2'
+    hooks:
+      - id: ruff
+        args: ["--fix"]
+      - id: ruff-format
+  # - repo: https://github.com/numpy/numpydoc
```

Comment:
Even after excluding `test.*` items this shows 2884 failures. My vote is to enable and attempt to address these in a follow up PR.

Reply:
Sounds fine to me

Reply:
Issue opened to track this: https://github.com/spacetelescope/stdatamodels/issues/385